lib/spack/spack/binary_distribution.py

```diff
@@ -942,32 +955,125 @@ def _read_specs_and_push_index(file_list, cache_prefix, db, db_root_dir):
     )
 
 
-def generate_package_index(cache_prefix):
-    """Create the build cache index page.
+def _specs_from_cache_aws_cli(cache_prefix):
+    """
+    Use aws cli to sync all the specs into a local temp directory, returning
+    the list of local file paths and a function that can read each one from
+    the file system.
+    """
```

Comment:
Can we use Google style for docstrings? That would be:
```suggestion
    """Use aws cli to sync all the specs into a local temporary directory.
    
    Args:
        cache_prefix (str): prefix of the build cache on s3
        
    Return:
        List of the local file paths and a function that can read each one from the file system.
    """
```